lib/batches/batch_spec.go

```diff
@@ -199,9 +199,22 @@ func parseBatchSpec(schema string, data []byte, opts ParseBatchSpecOptions) (*Ba
 		}
 	}
 
+	for i, step := range spec.Steps {
+		for _, mount := range step.Mount {
+			if strings.Contains(mount.Path, invalidMountCharacters) {
+				errs = errors.Append(errs, NewValidationError(errors.Newf("step %d mount path contains invalid characters", i+1)))
```

Comment:
Perhaps it'd be useful to show the invalid characters in the error message, too, so that it'd be easy for a user to fix?

```suggestion
				errs = errors.Append(errs, NewValidationError(errors.Newf("step %d mount path contains invalid characters (%s)", i+1, invalidMountCharacters)))
```